ambry-commons/src/test/java/com.github.ambry.commons/BlobIdTest.java

```diff
@@ -13,119 +13,440 @@
  */
 package com.github.ambry.commons;
 
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.clustermap.ClusterMap;
 import com.github.ambry.clustermap.MockClusterMap;
 import com.github.ambry.clustermap.MockPartitionId;
 import com.github.ambry.clustermap.PartitionId;
 import com.github.ambry.utils.ByteBufferInputStream;
 import java.io.DataInputStream;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Random;
 import java.util.UUID;
 import org.apache.commons.codec.binary.Base64;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+import static com.github.ambry.clustermap.ClusterMapUtils.*;
+import static com.github.ambry.commons.BlobId.*;
+import static com.github.ambry.utils.Utils.*;
 import static org.junit.Assert.*;
 
 
+/**
+ * Unit tests for {@link BlobId}. Using {@link BlobIdV2} builds a {@link BlobId}
+ * in v2 equivalent to setting {@link BlobId#CURRENT_VERSION} to be {@link BlobId#BLOB_ID_V2}.
+ */
+@RunWith(Parameterized.class)
 public class BlobIdTest {
```

Comment:
This is truly a useful test, and it is in Line 291
`assertEquals("Wrong base-64 ID in blobId: " + blobId, srcBlobIdStr, blobId.getID());`, where the `srcBlobIdStr` is a serialized string, and `blobId.getID()` is a string after deser-ser.